### DISCUSSION TITLE
sloc.xyz: exclude-dir support for category=code

### DISCUSSION THREAD

**Discussion by reinder** [boyter/scc#324] [Ideas]
Hi,

I'd love to add a *code* badge to my project (fantastic service!), however the repo also contains some code I'd like to exclude from the line count. Support for the `--exclude-dir` param would be great :)

Reinder

**Comment by ell1e** [comment]
Maybe it could also support the `.gitattributes` file with the attribute types `linguist-vendored` and `linguist-language=...`? Both are used to set or override the ignored folders as well as override the language detection if needed on github. (The latter can be useful not only if the detection unexpectedly fails but also if the repo contains an obscure language that can't be detected at all, but that happens to be identifiable via a unique file extension.)

This would allow settings already added for GitHub to carry right over.

**Comment by boyter** [comment]
I think before I do this I might need to rewrite the service. Its not very "cheap" to run at the moment and I would like to resolve that. I can then start fixing things like this at the same time. Ill see about doing it sometime in the next few weeks.
